### PR TITLE
Make hdfs block size configurable for hive external table

### LINKED ISSUE #977: Some HdfsScanNode are running slowly on accessing external hive table.
## Enhancement

There are some stragglers of hdfs scan node when accessing external hive table.

### case 1
```
HDFS_SCAN_NODE (id=0):(Active: 32s61ms[32061301213ns], % non-child: 99.87%)
   - BytesReadRemote: 32.56 MB
   - IoCounter: 2.641K (2641)
   - IoTime: 31s75ms
HDFS_SCAN_NODE (id=0):(Active: 2s590ms[2590921852ns], % non-child: 8.07%)
   - BytesReadRemote: 2.19 MB
   - IoCounter: 221
   - IoTime: 2s524ms
```

### case 2

```
HDFS_SCAN_NODE (id=0):(Active: 31s944ms[31944642297ns], % non-child: 98.47%)
   - BytesReadRemote: 582.40 MB
   - IoCounter: 5.841K (5841)
   - IoTime: 1m44s
HDFS_SCAN_NODE (id=0):(Active: 31s752ms[31752363173ns], % non-child: 97.88%)
   - BytesReadRemote: 534.62 MB
   - IoCounter: 3.127K (3127)
   - IoTime: 1m28s
HDFS_SCAN_NODE (id=0):(Active: 24s129ms[24129090498ns], % non-child: 74.38%)
   - BytesReadRemote: 551.62 MB
   - IoCounter: 5.941K (5941)
   - IoTime: 1m26s
HDFS_SCAN_NODE (id=0):(Active: 6s424ms[6424808934ns], % non-child: 19.80%)
   - BytesReadRemote: 541.38 MB
   - IoCounter: 3.346K (3346)
   - IoTime: 1m3s
HDFS_SCAN_NODE (id=0):(Active: 4s913ms[4913008130ns], % non-child: 15.14%)
   - BytesReadRemote: 535.38 MB
   - IoCounter: 3.12K (3120)
   - IoTime: 54s43ms
HDFS_SCAN_NODE (id=0):(Active: 4s671ms[4671950014ns], % non-child: 14.40%)
   - BytesReadRemote: 552.49 MB
   - IoCounter: 3.205K (3205)
   - IoTime: 1m3s
```

### PR DESCRIPTION
**Pull request by gengjun-git**
Add a new configuration hdfs_block_size. It's used to split hdfs files into blocks for hive external table. 
If you want to use the default block size of hdfs, set this parameter to -1.

### CODE REVIEW

**Review by dirtysalt — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaClient.java [hunk 1]
```diff
@@ -468,6 +468,21 @@ private double getLengthFromLiteral(LiteralExpr literalExpr, Type type) {
     }
 
     private List<HdfsFileDesc> getHdfsFileDescs(String dirPath) throws Exception {
+        // NOTE get the configure value to local variable, because the config value can be changed at runtime.
+        long hdfsBlockSize = Config.hdfs_block_size;
+        long osBlockSize = Config.object_storage_block_size;
+        if (!isObjectStorage(dirPath)) {
+            if (hdfsBlockSize < 0) {
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaClient.java:475]: hdfsBlockSize <= 0

**Review by chaoyli — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaClient.java (hunk 1 above)
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/external/hive/HiveMetaClient.java:475]: hdfsBlockSize <= 0

**Review by chaoyli — CHANGES_REQUESTED** [review]

**Comment by chaoyli** [comment]
The hdfs_block_size is mislead. I will think the StarRocks intends to configure the block size of hdfs cluster.

**Comment by gengjun-git** [comment]
same to https://github.com/StarRocks/starrocks/pull/1173

**Comment by gengjun-git** [comment]
same with https://github.com/StarRocks/starrocks/pull/1173

**Pull request closed without merging**